src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -1,32 +1,122 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
 using System.CommandLine.Parsing;
+using System.IO;
 using System.Linq;
+using System.Text.Json;
 using Microsoft.DotNet.Cli;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
 
 namespace Microsoft.DotNet.Workloads.Workload.Install
 {
     internal class WorkloadInstallCommand : CommandBase
     {
-        private readonly string _framework;
-        private IReadOnlyCollection<string> _workloadIds;
+        private readonly string _fromCacheOption;
+        private readonly bool _printDownloadLinkOnly;
+        private readonly IReadOnlyCollection<string> _workloadIds;
+
+        public readonly string MockInstallDirectory = Path.Combine(CliFolderPathCalculator.DotnetUserProfileFolderPath,
+            "DEV_mockworkloads");
 
         public WorkloadInstallCommand(
             ParseResult parseResult)
             : base(parseResult)
         {
-            _framework = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FrameworkOption);
-            _workloadIds = parseResult.ValueForArgument<IReadOnlyCollection<string>>(WorkloadInstallCommandParser.WorkloadIdArgument);
+            _workloadIds =
+                parseResult.ValueForArgument<IReadOnlyCollection<string>>(WorkloadInstallCommandParser
+                    .WorkloadIdArgument);
+            _printDownloadLinkOnly =
+                parseResult.ValueForOption<bool>(WorkloadInstallCommandParser.PrintDownloadLinkOnlyOption);
+            _fromCacheOption = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FromCacheOption);
         }
 
         public override int Execute()
         {
             // TODO stub
-            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ",_workloadIds)}");
+            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ", _workloadIds)}");
+            List<string> allowedMockWorkloads = new List<string> {"mobile-ios", "mobile-android"};
+
+            if (!_printDownloadLinkOnly && string.IsNullOrWhiteSpace(_fromCacheOption))
+            {
+                Reporter.Output.WriteLine(
```

Comment:
The workload install logic I'm working on now will conflict with all of this. Would it make sense to put this behind a command line flag? Or is that the intention here, and the "real" install logic will be integrated when `!_printDownloadLinkOnly && string.IsNullOrWhiteSpace(_fromCacheOption)` is true?

Reply:
Once you have the proper implementation, you can delete all these. VSMac should be able to code again a certain build with mocks instead of keeping these mock working